### PR TITLE
DROID-2734 Global search | Save last search phrase

### PR DESCRIPTION
**Pull request by konstantiniiv**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: persistence/src/main/java/com/anytypeio/anytype/persistence/repo/DefaultUserSettingsCache.kt [hunk 1]
```diff
@@ -300,6 +300,54 @@ class DefaultUserSettingsCache(
...
+
+    override suspend fun getLastSearchQuery(space: SpaceId): String {
+        return context.spacePrefsStore
+            .data
+            .map { preferences ->
+                preferences
+                    .preferences[space.id]
+                    ?.lastSearchQuery.orEmpty()
+            }
+            .first()
```
Comment (uburoiubu) [inline persistence/src/main/java/com/anytypeio/anytype/persistence/repo/DefaultUserSettingsCache.kt:330]: `firstOrNull` might be safier

File: presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt [hunk 2]
```diff
@@ -195,11 +200,12 @@ class GlobalSearchViewModel(
                 keys = DEFAULT_KEYS,
                 filters = ObjectSearchConstants.filterSearchObjects(
                     // TODO add tech space?
-                    spaces = listOf(spaceManager.get())
+                    spaces = listOf(space.id)
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/search/GlobalSearchViewModel.kt:203]: yes!

**Review by uburoiubu — APPROVED** [review]

**Comment by uburoiubu** [comment]
well done

File: persistence/src/main/java/com/anytypeio/anytype/persistence/repo/DefaultUserSettingsCache.kt (hunk 1 above)
Comment (konstantiniiv) [reply persistence/src/main/java/com/anytypeio/anytype/persistence/repo/DefaultUserSettingsCache.kt:330]: the error is handled everywhere, so this is the expected behavior

**Pull request merged**